### ISSUE TITLE
Consistency with newlines in object literals

### ISSUE THREAD

**Issue by dburles** [prettier/prettier#960]
Sorry if this has been brought up before, I couldn't seem to find anything on it.

Take a look at this example: https://prettier.github.io/prettier/#%7B%22content%22%3A%22const%20example%20%3D%20%7B%5Cn%20%20one%3A%201%2C%5Cn%20%20two%3A%202%2C%5Cn%20%20three%3A%203%2C%5Cn%7D%3B%5Cn%5Cnconst%20example2%20%3D%20%7B%20one%3A%201%2C%20two%3A%202%2C%20three%3A%203%20%7D%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22all%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22doc%22%3Afalse%7D%7D

I would expect them to both be formatted *one way* and in this case since it doesn't exceed the maximum line length I'd assume they should both appear on a single line like `example2`.

I find I end up with inconsistency in my own code because of this.

**Comment by vjeux** [comment]
Hey @dburles, sorry we haven't made a good explanation for this one. The issue with objects is that there are many use cases where you want them expanded and many use cases where you want them collapsed. Unfortunately, we haven't found a good automatic way to decide which is which.

So, we decided to give some power to the developer always keeping an object expanded if it was expanded in the first place. This is definitely not ideal and causes the kind of issues you are seeing here, but this is the best heuristic that we found.

In order for both to match, you either have to remove all \n from both, or put a \n anywhere in both and reformat your code.
